### PR TITLE
Fix standalone password generator so that the esc key quits out of it

### PR DESCRIPTION
**Pull request by pkill-9**
## Description
In the password generator (standalone, ie not when you're editing an entry) the ESC key is ignored, however in most other dialogs ESC is a shortcut to quit the dialog.

## Motivation and context
Reported in issue 953

## Types of changes
- ✅ Bug fix (non-breaking change which fixes an issue)

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**

### CODE REVIEW

**Comment by pkill-9** [comment]
Reading the logic, I don't believe this does make Ctrl-Enter a close shortcut, it just limits when ESC acts as a close shortcut depending on whether modifier keys are also pressed. I added the handling of modifier keys since that's the same behavior for other dialogs (as defined in DialogyWidget.cpp)

What do you mean by adding a keyboard shortcut? (Is that what's done in DialogyWidget?) If that would be a better approach I can definitely explore it further.

(My apologies if my knowledge is limited, this is a first foray into this project so I'm still learning how everything fits together)

File: src/gui/PasswordGeneratorWidget.cpp [hunk 1]
```diff
@@ -149,6 +150,20 @@ void PasswordGeneratorWidget::setStandaloneMode(bool standalone)
...
+void PasswordGeneratorWidget::keyPressEvent(QKeyEvent* e)
+{
+    if (!e->modifiers() || (e->modifiers() & Qt::KeypadModifier && e->key() == Qt::Key_Enter)) {
+        if (e->key() == Qt::Key_Escape && m_standalone == true) {
+            emit dialogTerminated();
+        } else {
+            e->ignore();
+        }
+    }
+    else {
```
Comment (phoerious) [inline src/gui/PasswordGeneratorWidget.cpp]: Please put this on the same line as the closing brace. Other than that, I think this patch is fine as is.

**Comment by TheZ3ro** [comment]
@pkill-9 can you rebase on develop?

**Comment by TheZ3ro** [comment]
Seems that this PR broke some time ago. Submitting a new PR #1252 with cherry-picked commit (since I can't push to this pr :(

**Pull request closed without merging**